django/src/rdwatch/management/commands/ingest.py

```diff
@@ -0,0 +1,258 @@
+
+def tile_raster(rst: GDALRaster) -> Iterable[GDALRaster]:
+    if not len(rst.bands) == 1:
+        raise ValueError("Expected single-band raster")
+    if not rst.srid == 3857:
+        raise ValueError("Expected raster projected to SRID 3857")
+
+    bnd = rst.bands[0]
+    tilesize = 64
+
+    for xoffset in range(0, bnd.width, tilesize):
+        for yoffset in range(0, bnd.height, tilesize):
+            size = (
+                tilesize if xoffset + tilesize <= rst.width else rst.width - xoffset,
+                tilesize if yoffset + tilesize <= rst.height else rst.height - yoffset,
+            )
+            data = bnd.data(
+                offset=(xoffset, yoffset),
+                size=size,
+            )
+            yield GDALRaster(
+                {
+                    "width": tilesize,
+                    "height": tilesize,
+                    "scale": rst.scale,
+                    "origin": [
+                        rst.origin.x + xoffset * rst.scale.x,
+                        rst.origin.y + yoffset * rst.scale.y,
+                    ],
+                    "skew": rst.skew,
+                    "srid": rst.srid,
+                    "nr_of_bands": 1,
+                    "datatype": bnd.datatype(),
+                    "bands": [
+                        {
+                            "data": data,
+                            "size": size,
+                            "nodata_value": bnd.nodata_value,
+                        }
+                    ],
+                }
+            )
+
+
+def _ingest_geojson(
+    site_geojson_file: Path,
+    prediction_configuration: PredictionConfiguration,
+    tracking_configuration: TrackingConfiguration,
+) -> list[Site]:
+    """
+    Ingest the geojson file at the given path into the Site model,
+    returning a list of Site instances.
+    """
+    site_geojson = json.loads(site_geojson_file.read_text())
+
+    sites: list[Site] = []
+
+    # Ingest each site in the geojson
+    for site in site_geojson["features"][1:]:
+        properties: dict = site["properties"]
+
+        label = label_mapping[properties["current_phase"]]
+
+        score = properties["score"]
+
+        geometry = GEOSGeometry(json.dumps(site["geometry"]))
+        assert isinstance(geometry, MultiPolygon | Polygon)
+
+        # Generate a MultiPolygon if the geometry is just a Polygon (this
+        # seems to be the case for all ground truth features).
+        if isinstance(geometry, Polygon):
+            geometry = MultiPolygon(geometry)
+
+        # Parse the crop string "identifier" from the parent
+        # directory of the source TIF given in the geojson.
+        crop_string = Path(properties["source"]).parent.name
+
+        # Parse the crop string, extracting the metadata encoded in it
+        crop_parse = CropParse.from_string(crop_string)
+
+        sensor_name = crop_parse.sensor
+
+        assert sensor_name
+        sensor_name = "L8" if sensor_name == "Landsat 8" else "S2"
```

Comment:
I think `crop_parse.sensor` will already be `"L8"` or `"S2"`, not `"Landsat 8"` or `"Sentinel 2"`.